torch/distributed/pipelining/_backward.py

```diff
@@ -189,35 +189,30 @@ def hook(grad_inputs):
             handle = intermediate.register_prehook(get_hook(param_group, i))
             handles.append(handle)
 
-    # Stage 0 inputs do not require grads? Should we skip in that case?
-    if all(tensor.requires_grad for tensor in input_values):
```

Comment:
how come this if condition can be removed now?

Reply:
we no longer call `stage_backward_input` for the first stage anymore